apt_installer/generate_dockerfile.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/ghodss/yaml"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+)
+
+var INTERMEDIATE_IMAGE_TAG = "apt:intermediate"
+
+func generateDockerfile(installer Installer) (error, string) {
+	var err error
+
+	build_dir, err := ioutil.TempDir(".", "intermediate_base.")
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error when creating temp_dir: %s", err)), ""
+	}
+
+	f, err := os.Create(path.Join(build_dir, "/Dockerfile"))
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error opening file for writing: %s", err)), build_dir
+	}
+
+	defer f.Close()
+	w := bufio.NewWriter(f)
+
+	if err := DOCKERFILE_TMPL.Execute(w, installer); err != nil {
+		return errors.New(fmt.Sprintf("Error when executing template: %s", err)), build_dir
+	}
+
+	w.Flush()
+	return nil, build_dir
+}
+
+func doBuild(build_dir string) error {
+	docker_flags := []string{"build", "--no-cache", "-t", INTERMEDIATE_IMAGE_TAG, build_dir}
```

Comment:
I originally had this image built from scratch, which was great, but I realized that since the image needs to be built after generating the Dockerfile, I either bundle it in with this image or require that yet another build step is added to the cloudbuild whenever someone wants to use this. What do you think?